docs/design.md

```diff
@@ -1,92 +1,122 @@
-+ **Logic**: The main logic of the app is in Java POJOs (Plain Old Java Objects). Some automated tasks are implemented as Servlets.
-+ **Storage**: Storage uses the persistence framework provided by GAE 'datastore', a noSQL database.
-+ **Test Driver**: TEAMMATES makes heavy use of automated regression testing. TestNG is used for Java unit testing and QUnit for JavaScript unit testing. We use Selenium Web Driver to automate UI testing. Test Driver can access the application directly to set up test data. We use JSON format to transmit such data to the application.
-+ **Client**: The Client component can connect to the back end directly without using a Browser. It is used for administrative purposes. e.g., migrating data to a new schema
-+ **Common**: The Common component contains utility code used across the application.
+TEAMMATES is a Web application that runs on Google App Engine (GAE) cloud platform. Given above is an overview of the main components.
+- **UI**: The UI seen by users consists of Web pages containing HTML, CSS (for styling) and JavaScript (for client-side interactions such as sorting, input validation etc.). This UI is generated by Java Server Pages (JSP), using the JSP Standard Tag Library (JSTL), and Java Servlets residing on the server. Requests are sent to the server over HTTP. In addition, AJAX (Asynchronous JavaScript And XML) is used (sparingly) to fetch data from the server asynchronously.
+- **Logic**: The main logic of the application is in POJOs (Plain Old Java Objects).
+- **Storage**: The storage layer of the application uses the persistence framework provided by **GAE Datastore**, a NoSQL database.
+- **Test Driver**: TEAMMATES makes heavy use of automated regression testing. TestNG is used for Java testing (all levels) and QUnit for JavaScript unit-testing. Selenium WebDriver is used to automate UI testing with actual Web browsers. This component can access the application directly to set up test data (transmitted using JSON format).
+- **Client**: The Client component can connect to the back end directly without using a Web browser. It is used for administrative purposes, e.g. migrating data to a new schema.
+- **Common**: The Common component contains utility code (data transfer objects, helper classes, etc.) used across the application.
 
 The diagram below shows how the code is organized into packages inside each component and dependencies among them.
 
 ![Package Diagram](images/packageDiagram.png)
 
 Notes:
-+ [logic] - [ui::view] - [ui::controller] represent an application of Model-View-Controller pattern.
-+ ui::view and ui::website packages are not Java packages. They consist of jsp, html, js and css files.
+- `[logic] - [ui::view] - [ui::controller]` represent an application of Model-View-Controller pattern.
 
-##UI
+## UI Component
 
 The diagram below shows the object structure of the UI component.
 
 ![UI Component](images/UiComponent.png)
```

Comment:
Can I request to hold these for now? Three reasons:
- The UI component diagram is crowded enough as it is, and adding `HttpServletResponse` to the picture (pun unintended) is not going to help.
- The pathway for `HttpServletRequest` follows from what it looks like in the previous UI component diagram. While I believe that your claims are right in some parts, these are some black-boxed, more fine-tuned details that require deeper understanding of the system. This entails two things:
  - We both may be wrong even if we're confident we're not
  - The detailed detail is not so important for the readers of this document
- We can be purists in another issue/PR, and merge this one soon 😛